example/cpp/docs/decimal_ffi.rst

```diff
@@ -0,0 +1,80 @@
+``decimal::ffi``
+================
+
+
+.. cpp:class:: ICU4XFixedDecimalFormat
+
+
+    An ICU4X Fixed Decimal Format object, capable of formatting a :cpp:class:`ICU4XFixedDecimal` as a string. See `the Rust docs <https://unicode-org.github.io/icu4x-docs/doc/icu/decimal/struct.FixedDecimalFormat.html>`__ for more information.
+
+    .. cpp:function:: static ICU4XFixedDecimalFormatResult try_new(const ICU4XLocale& locale, const ICU4XDataProvider& provider, ICU4XFixedDecimalFormatOptions options)
+
+        Creates a new :cpp:class:`ICU4XFixedDecimalFormat` from locale data. See `the Rust docs <https://unicode-org.github.io/icu4x-docs/doc/icu/decimal/struct.FixedDecimalFormat.html#method.try_new>`__ for more information.
+
+    .. cpp:function:: void format_write(const ICU4XFixedDecimal& value)
+
+        Formats a :cpp:class:`ICU4XFixedDecimal` to a string. See `the Rust docs <https://unicode-org.github.io/icu4x-docs/doc/icu/decimal/struct.FixedDecimalFormat.html#method.format>`__ for more information.
+
+.. cpp:struct:: ICU4XFixedDecimalFormatOptions
+
+
+
+    .. cpp:member:: ICU4XFixedDecimalGroupingStrategy grouping_strategy
+
+
+
+    .. cpp:member:: ICU4XFixedDecimalSignDisplay sign_display
+
+
+
+    .. cpp:function:: static ICU4XFixedDecimalFormatOptions default_()
+
+
+.. cpp:struct:: ICU4XFixedDecimalFormatResult
+
+
+
+    .. cpp:member:: std::optional<ICU4XFixedDecimalFormat> fdf
+
+
+        The :cpp:class:`ICU4XFixedDecimalFormat`, exists if creation was successful.
+
+    .. cpp:member:: bool success
+
+
+        Whether creating the :cpp:class:`ICU4XFixedDecimalFormat` was successful.
+
+.. cpp:enum-struct:: ICU4XFixedDecimalGroupingStrategy
+
```

Comment:
question: What's up with the extra whitespace?

Reply:
There aren't any docs, so you get a bunch of extra lines around where the docs would have been.

Reply:
Ah. Would be nice if we can elide that

Reply:
fixed!

Reply:
I think part of my issue is that at least for enum variants we should probably squish them all together. But it doesn't matter that much, this isn't human-read!